apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/next/ast/CodeGenerationAstBuilder.kt

```diff
@@ -0,0 +1,286 @@
+package com.apollographql.apollo.compiler.next.ast
+
+import com.apollographql.apollo.api.internal.QueryDocumentMinifier
+import com.apollographql.apollo.compiler.OperationIdGenerator
+import com.apollographql.apollo.compiler.ast.CustomTypes
+import com.apollographql.apollo.compiler.escapeKotlinReservedWord
+import com.apollographql.apollo.compiler.ir.CodeGenerationIR
+import com.apollographql.apollo.compiler.ir.Operation
+import com.apollographql.apollo.compiler.ir.TypeDeclaration
+import com.apollographql.apollo.compiler.parser.introspection.IntrospectionSchema
+import com.apollographql.apollo.compiler.parser.introspection.resolveType
+import com.apollographql.apollo.compiler.ir.Fragment as IrFragment
+
+internal fun CodeGenerationIR.buildCodeGenerationAst(
+    schema: IntrospectionSchema,
+    customTypeMap: CustomTypes,
+    typesPackageName: String,
+    fragmentsPackage: String,
+    useSemanticNaming: Boolean,
+    operationIdGenerator: OperationIdGenerator
```

Comment:
That will most likely conflict with https://github.com/apollographql/apollo-android/pull/2476 since it takes a `operationOutput` instead of an `operationIdGenerator` but I can rebase once this is merged.